src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1164,6 +1164,9 @@ await notificationService.PublishUpdateAsync(cr.ModelResource, s => s with
 
     private async Task CreateExecutableAsync(AppResource er, ILogger resourceLogger, CancellationToken cancellationToken)
     {
+        var connectionStringAvailableEvent = new ConnectionStringAvailableEvent(er.ModelResource, serviceProvider);
```

Comment:
these events should only fire for IResourceWithConnectionString

Reply:
There is a corner case here that we'll need to explore around project to project references where we might have a health check in the apphost which checks for the liveness of the projects endpoints.

What I am thinking is that we do something like this:

```csharp
var api = builder.AddProject<Projects.BackendApi>("backendapi")
                 .WithEndpoint("https", (ep) => ep.HealthCheck = true);
var frontend = builder.AddProject<Projects.Frontend>("frontend")
                      .WithReference(api).WaitFor(api);
```

When you enable health checks on an endpoint we would automatically inject a HTTP health check for /healthz (or something else) and the frontend project in this case wouldn't start until the backend API was healthy.

Reply:
... I don't see this as critical right now but its what I've been thinking about to approach this if it comes up as a future requirement.

Reply:
Are we bubbling the state down to the child resource by default? If I wait for the db what happens right now?

Reply:
This comment might have been in relation to another comment. Regardless ...

Right now the child resource picks up the state of the parent (e.g. it'll move into the Running state when the parent does). But it doesn't have any health checks that might block something that depends on the database from launching.

If we do add a health check to the database resource we'll need to fire before resource started events for child resources artificially. Just need to try it out and find the potential places where it could result in a hang.

Reply:
Got child resource waits working.